newa/cli.py

```diff
@@ -291,8 +302,12 @@ def cmd_schedule(ctx: CLIContext) -> None:
 def cmd_execute(ctx: CLIContext, workers: int) -> None:
     ctx.enter_command('execute')
 
-    if not os.environ.get("TESTING_FARM_API_TOKEN"):
+    tf_token = ctx.settings.get("tf_token")
+    if not tf_token:
         raise ValueError("TESTING_FARM_API_TOKEN not set!")
+    # make some variables available for workers
+    os.environ["TESTING_FARM_API_TOKEN"] = tf_token
+    os.environ["NEWA_TF_RECHECK_DELAY"] = ctx.settings.get("tf_recheck_delay")
```

Comment:
This does not look right. Code takes values from loades settings, puts them into environment, and the worker code then ignores settings and peeks into environment for *some* keys.

Would you mind checking `Pool.starmap()` instead of `Pool.map()`? That should let us pass more arguments to `worker()`, and we could propagate `ctx.settings`. Something like this should work:

```python
schedule_list = [
  (ctx, ctx.state_dirpath / child.name)
  for child in ctx.state_dirpath.iterdir()
  if child.name.startswith('schedule-')]

...

for _ in worker_pool.starmap(worker, schedule_list):

...

def worker(ctx: CLIContext, schedule_file: Path) -> None:
    ...
    delay = int(ctx.settings.tf_recheck_delay)
```

Reply:
Fixed. Thank you, I was not aware of `starmap`.